pkg/core/basic_chain_test.go

```diff
@@ -596,3 +625,1634 @@ func TestBlockchain_StartFromExistingDB(t *testing.T) {
+					txSetOracle.Scripts = []transaction.Witness{{
+						InvocationScript:   testchain.SignCommittee(txSetOracle),
+						VerificationScript: testchain.CommitteeVerificationScript(),
+					}}
+					bl := block.New(bc.config.StateRootInHeader)
+					bl.Index = bc.BlockHeight() + 1
+					ic := bc.newInteropContext(trigger.All, bc.dao, bl, txSetOracle)
+					ic.SpawnVM()
+					ic.VM.LoadScript([]byte{byte(opcode.RET)})
+					require.NoError(t, bc.contracts.Designate.DesignateAsRole(ic, noderoles.Oracle, oraclePubs))
+					_, err = ic.DAO.Persist()
+					require.NoError(t, err)
+
+					t.Run("Valid", func(t *testing.T) {
+						tx := getOracleTx(t)
+						require.NoError(t, oracleAcc.SignTx(netmode.UnitTestNet, tx))
+						require.NoError(t, bc.VerifyTx(tx))
+
+						t.Run("NativeVerify", func(t *testing.T) {
+							tx.Signers = append(tx.Signers, transaction.Signer{
+								Account: bc.contracts.Oracle.Hash,
+								Scopes:  transaction.None,
+							})
+							tx.Scripts = append(tx.Scripts, transaction.Witness{})
+							t.Run("NonZeroVerification", func(t *testing.T) {
+								w := io.NewBufBinWriter()
+								emit.Opcodes(w.BinWriter, opcode.ABORT)
+								emit.Bytes(w.BinWriter, util.Uint160{}.BytesBE())
+								emit.Int(w.BinWriter, 0)
+								emit.String(w.BinWriter, orc.Manifest.Name)
+								tx.Scripts[len(tx.Scripts)-1].VerificationScript = w.Bytes()
+								err := bc.VerifyTx(tx)
+								require.True(t, errors.Is(err, ErrNativeContractWitness), "got: %v", err)
+							})
+							t.Run("Good", func(t *testing.T) {
+								tx.Scripts[len(tx.Scripts)-1].VerificationScript = nil
+								require.NoError(t, bc.VerifyTx(tx))
+							})
+						})
+					})
+					t.Run("InvalidRequestID", func(t *testing.T) {
+						tx := getOracleTx(t)
+						tx.Attributes[0].Value.(*transaction.OracleResponse).ID = 2
+						require.NoError(t, oracleAcc.SignTx(netmode.UnitTestNet, tx))
+						checkErr(t, ErrInvalidAttribute, tx)
+					})
+					t.Run("InvalidScope", func(t *testing.T) {
+						tx := getOracleTx(t)
+						tx.Signers[0].Scopes = transaction.Global
+						require.NoError(t, oracleAcc.SignTx(netmode.UnitTestNet, tx))
+						checkErr(t, ErrInvalidAttribute, tx)
+					})
+					t.Run("InvalidScript", func(t *testing.T) {
+						tx := getOracleTx(t)
+						tx.Script = append(tx.Script, byte(opcode.NOP))
+						require.NoError(t, oracleAcc.SignTx(netmode.UnitTestNet, tx))
+						checkErr(t, ErrInvalidAttribute, tx)
+					})
+					t.Run("InvalidSigner", func(t *testing.T) {
+						tx := getOracleTx(t)
+						tx.Signers[0].Account = accs[0].Contract.ScriptHash()
+						require.NoError(t, accs[0].SignTx(netmode.UnitTestNet, tx))
+						checkErr(t, ErrInvalidAttribute, tx)
+					})
+					t.Run("SmallFee", func(t *testing.T) {
+						tx := getOracleTx(t)
+						tx.SystemFee = 0
+						require.NoError(t, oracleAcc.SignTx(netmode.UnitTestNet, tx))
+						checkErr(t, ErrInvalidAttribute, tx)
+					})
+				})
+		*/
```

Comment:
Everything else is pretty simple. I don't think you need a properly running service for this test.

Reply:
Fixed.